### ISSUE TITLE
Use autocomplete to add component's control statement when authoring component control statements

### ISSUE THREAD

**Issue by gregelin** [GovReady/govready-q#1066]
As a GovReady user,I want to be able to be able to choose a component via autocomplete/drop-down when adding a component-control statement in the control authoring interface,so I can get the official name of the component and pre-populate the control description.

# Criteria
## Essential
* Autocomplete of components
* Being able to type-in or add a NEW component
* Populate the related statement(s) for active control from a component's related, control-specific `control_implementation_prototype` statement(s)

## Better
* Gracefully handle adding _all_ of selected component's controls to the system. This might be done by a pop-up notification telling user that all controls related to selected component have been added. May be give user option to only populate single statement(s) for active control?

## Best
* instead of just component names, additional information is available about popup
* Combine new, select existing, autocomplete, and drop down all in one widget box. For example, how AirTable adds related records.

# Screenshots
**Control authoring environment screenshot where autocomplete needed**
<img width="1105" alt="Screen Shot 2020-11-01 at 10 57 43 AM" src="https://user-images.githubusercontent.com/24979/97807928-998c0e00-1c31-11eb-86b3-c51c62e4b032.png">

**Selected components screenshot showing an implemented autocomplete**
<img width="1070" alt="Screen Shot 2020-11-01 at 11 02 28 AM" src="https://user-images.githubusercontent.com/24979/97807983-e4a62100-1c31-11eb-9855-9be4cf8934cf.png">

**Comment by gregelin** [comment]
@davidpofo Note on technical implementation.

An example autocomplete for components can be found `templates/systems/components_selected.html` using the `js/select2.min.js` library which combines dropdown box with autocomplete. It would be good to try and apply the same method of autocomplete in `templates/controls/editor.html` template. 

Snippet for how the `js/select2.min.js`-based autocomplete form is rendered.
```
    <div id="tab-content" class="row row-control" style="">
      <form class="form-inline" method="post" action="/systems/{{ system.id }}/components/add_system_component">
        {% csrf_token %}
        <div id="" class="col-xs-8 col-sm-8 col-md-8 col-lg-8 col-xl-8">
        <div class="form-group">
            <label for="compoment">Add component</label>&nbsp;&nbsp;
              <select class="producer_element_id" id="producer_element_id" name="producer_element_id" style="width:340px;">
                {% for component in elements %}
                <option value="{{ component.id }}">{{ component.name }}</option>
                {% endfor %}
              </select>
        </div>
          &nbsp;&nbsp;<button type="submit" class="small btn btn-sm btn-success" style="color: white;" onclick="add_smt()">Add</button>
      </div>
    </div>
      </form>
  </div>
```

Snippet for how the Javascript handles the autocomplete:
```
<script src="{% static "js/select2.min.js" %}"></script>
<link href="{% static "css/select2.min.css" %}" rel="stylesheet" />

<script type="text/javascript">
   function show_control_by_id() {
     var control_id = $('#control_lookup').find('input[name="id"]').val();
     var url = "/controls/"+"800-53"+"/"+control_id+"/";
     window.location.href = url;
   }

  $(document).ready(function() {
    $('.producer_element_id').select2();
  });
 </script>
```

Note that autocomplete for selecting a component in `templates/controls/editor.html` should probably only be added  for the version of the component-statement form that lives as a HTML snippet inside the JavaScript for the page. This HTML snippet is used to insert the \ additional HTML form for a _new_ component-control statement into the HTML document object.. There are minor differences between the form for authoring a new component control statement (found in the Javascript) and the form for editing existing component-control statement. The form for editing an existing component-statement is tied to a component that exists in the database. The form for creating a new component-statement allows for using the NAME of the component to match against an existing component in the database to associate the statement with an existing component, or create a new component to be created based on the name of the component and associate the statement with that newly created component.

If we are using autocomplete to add a component-statement, it is possible for the user to select a component that _already_ exists in the database _and_ has an existing `control_implementation_prototype` available for the component for the active control. 

In other words, when a user adds a component-statement control to a control for the system there are three cases:
1. User is adding a new component and a new component-statement
2. User is selecting an existing component and getting a copy of the existing component-statement for that control (e.g., copy of what is in `control_implementation_prototoype` for that component and statement
3. User is selecting an existing component and authoring a _new_ component-statement for that component for that control

**Comment by gregelin** [comment]
An alternative autocomplete library suggested by @davidpofo:

https://django-autocomplete-light.readthedocs.io/en/3.5.1/tutorial.html
